pages/add-content.js

```diff
@@ -0,0 +1,58 @@
+import { useRouter } from "next/router";
+import { useState } from "react";
+import Form from "~/components/Form";
+import Layout from "~/components/Layout/Layout";
+
+const AddContentPage = () => {
+  const router = useRouter();
+  const [error, setError] = useState(false);
+
+  // send data obj to api/content
+  const handleAddContentSubmit = async (data) => {
+    setError(false);
+    const response = await fetch("/api/content", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+      },
+      body: JSON.stringify(data),
+    });
+
+    // if response ok redirect to issue page
+    if (response.ok) {
+      router.push(`/issues/${data.relations}`);
+    } else {
+      setError(true);
+    }
+  };
+
+  //data obj for the form
+  const data = {
+    title: "",
+    description: "",
+    content_type: "",
+    img_url: "",
+    video_url: "",
+    relations: "",
+  };
```

Comment:
Why declare this here? Why not declare it inside the <Form /> component to save having to pass it in as a prop? 

I'm potentially missing something here

Reply:
It's because I use the same form for updating and adding the content. 
So I need to pass the data object as a prop to the form.